### PR TITLE
reconfiguring the example file so that it builds and is easier to export

### PR DESCRIPTION
**Pull request by ebrillhart**
This reconfigures the file structure that holds the live Spectacle example and updates the build script to include it. This will ease the transfer of the example from its current location at stack.formidable.com to formidable.com/open-source/spectacle/example. Hopefully this works/the example was not previously excluded from the build for an important reason. 

I think an npm publish will be required once this is merged, and I'm happy to go through that process if pre-approved to do so!

cc/ @ryan-roemer @bmathews @kenwheeler

### CODE REVIEW

**Comment by ryan-roemer** [comment]
I think this needs updates to the ignore files as well. Gitignore and npmignore to be consistent with the existing source and lib directories

**Comment by ebrillhart** [comment]
@ryan-roemer like so?

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules
 dist
 lib
+example/lib
```
Comment (ebrillhart) [inline .gitignore]: @ryan-roemer like so?

Comment (bmathews) [inline .gitignore]: yeah, not needed, since the prior `lib` will cover it

File: .npmignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 node_modules
 src
+example/src
```
Comment (bmathews) [inline .npmignore]: npmignore follows same rules as gitignore. prior `src` will cover `example/src`, so this is not needed

**Comment by ebrillhart** [comment]
@ryan-roemer @bmathews good to go once I revert the gitignore and npmignore back?

File: .npmignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 node_modules
 src
-*.log
+*.log
```
Comment (ryan-roemer) [inline .npmignore]: Let's get that newline back too 😉

**Comment by ryan-roemer** [comment]
@ebrillhart -- One nit, after which LGTM. No re-review needed.

**Pull request merged**